build-scripts/config_common.cmake

```diff
@@ -248,6 +248,17 @@ if (WAMR_BUILD_SHARED_MEMORY EQUAL 1)
 else ()
   add_definitions (-DWASM_ENABLE_SHARED_MEMORY=0)
 endif ()
+if (WAMR_BUILD_MEMORY64 EQUAL 1)
+  # if native is 32-bit or cross-compiled to 32-bit
+  if (NOT (CMAKE_SIZEOF_VOID_P EQUAL 8 OR WAMR_BUILD_TARGET MATCHES ".*64.*"))
```

Comment:
This causes error not reported in Ubuntu x86-64 when building iwasm with `cmake .. -DWAMR_BUILD_TARGET=X86_32 -DWAMR_BUILD_MEMORY64=1`, using
`if (NOT WAMR_BUILD_TARGET MATCHES ".*64.*")` should be already good, since WAMR_BUILD_TARGET must have been set here (if it isn't set by developer, it will be set at the beginning of this file) and all 64-bit targets have "64" sub string (X86_64, AMD_64, AARCH64 and RISCV64).